src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,197 @@
+
+        } catch (HaltException halt) {
+            LoggerUtilImpl.getInstance().error(logger,
+                    String.format("cannot create event query: %1$s -> %2$s", halt.statusCode(), halt.body()), halt);
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.ERROR, "Given query was not found");
+        }
+
+        EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(eventQuery);
+        if (!feedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, String.format("cannot unregister event query: %1$s", feedback.getResponseText()));
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventQuery,
+                EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()))) {
+            halt(HttpStatusCodes.ERROR, "Could not delete query");
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String editEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        JsonObject jsonBody = new JsonObject();
+        try {
+            jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.BAD_REQUEST, "not a valid json");
+        }
+
+        JsonObject jsonEventQuery = jsonBody.get(JSON_EVENT_QUERY_ATTRIBUTE).getAsJsonObject();
+        if (jsonEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+        }
+
+        EventQuery newEventQuery = getEventQueryFromJson(jsonEventQuery);
+        if (newEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "failed to parse event query");
+        }
+
+        long oldEventQueryId = getEventQueryId(request);
+        EventQuery oldEventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(oldEventQueryId);
+        if (oldEventQuery == null) {
+            halt(HttpStatusCodes.ERROR, "Given query was not found");

```

Comment:
Please change the status code to `BAD_REQUEST`. Also use lower-case responses only